src/code/flg_set.c

```diff
@@ -1,9 +1,314 @@
+    { &gSaveContext.weekEventReg[68], "week_event_reg[68]" },
+    { &gSaveContext.weekEventReg[69], "week_event_reg[69]" },
+    { &gSaveContext.weekEventReg[70], "week_event_reg[70]" },
+    { &gSaveContext.weekEventReg[71], "week_event_reg[71]" },
+    { &gSaveContext.weekEventReg[72], "week_event_reg[72]" },
+    { &gSaveContext.weekEventReg[73], "week_event_reg[73]" },
+    { &gSaveContext.weekEventReg[74], "week_event_reg[74]" },
+    { &gSaveContext.weekEventReg[75], "week_event_reg[75]" },
+    { &gSaveContext.weekEventReg[76], "week_event_reg[76]" },
+    { &gSaveContext.weekEventReg[77], "week_event_reg[77]" },
+    { &gSaveContext.weekEventReg[78], "week_event_reg[78]" },
+    { &gSaveContext.weekEventReg[79], "week_event_reg[79]" },
+    { &gSaveContext.weekEventReg[80], "week_event_reg[80]" },
+    { &gSaveContext.weekEventReg[81], "week_event_reg[81]" },
+    { &gSaveContext.weekEventReg[82], "week_event_reg[82]" },
+    { &gSaveContext.weekEventReg[83], "week_event_reg[83]" },
+    { &gSaveContext.weekEventReg[84], "week_event_reg[84]" },
+    { &gSaveContext.weekEventReg[85], "week_event_reg[85]" },
+    { &gSaveContext.weekEventReg[86], "week_event_reg[86]" },
+    { &gSaveContext.weekEventReg[87], "week_event_reg[87]" },
+    { &gSaveContext.weekEventReg[88], "week_event_reg[88]" },
+    { &gSaveContext.weekEventReg[89], "week_event_reg[89]" },
+    { &gSaveContext.weekEventReg[90], "week_event_reg[90]" },
+    { &gSaveContext.weekEventReg[91], "week_event_reg[91]" },
+    { &gSaveContext.weekEventReg[92], "week_event_reg[92]" },
+    { &gSaveContext.weekEventReg[93], "week_event_reg[93]" },
+    { &gSaveContext.weekEventReg[94], "week_event_reg[94]" },
+    { &gSaveContext.weekEventReg[95], "week_event_reg[95]" },
+    { &gSaveContext.weekEventReg[96], "week_event_reg[96]" },
+    { &gSaveContext.weekEventReg[97], "week_event_reg[97]" },
+    { &gSaveContext.weekEventReg[98], "week_event_reg[98]" },
+    { &gSaveContext.weekEventReg[99], "week_event_reg[99]" },
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/flg_set/D_801DC120.s")
+    { &gSaveContext.eventInf[0], "event_inf[0]" },
+    { &gSaveContext.eventInf[1], "event_inf[1]" },
+    { &gSaveContext.eventInf[2], "event_inf[2]" },
+    { &gSaveContext.eventInf[3], "event_inf[3]" },
+    { &gSaveContext.eventInf[4], "event_inf[4]" },
+    { &gSaveContext.eventInf[5], "event_inf[5]" },
+    { &gSaveContext.eventInf[6], "event_inf[6]" },
+    { &gSaveContext.eventInf[7], "event_inf[7]" },
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/flg_set/func_800B3644.s")
+    { &gSaveContext.maskMaskBit[0], "mask_mask_bit[0]" },
+    { &gSaveContext.maskMaskBit[1], "mask_mask_bit[1]" },
+    { &gSaveContext.maskMaskBit[2], "mask_mask_bit[2]" },
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/flg_set/D_801DC9B8.s")
+    { NULL, NULL }, // used in the code to detect array end
+};
+
+static s32 entryIndex = 0;
+static u32 currentBit = 0;
+static s32 timer = 0;
```

Comment:
Since those are static, should be prefixed with `s`
```suggestion
static s32 sEntryIndex = 0;
static u32 sCurrentBit = 0;
static s32 sTimer = 0;
```